### PR TITLE
WIP: make configs round tripable

### PR DESCRIPTION
**Pull request by CJ-Wright**
So far this adds a failing test for round trip config.
I'm working on a fix.

### CODE REVIEW

**Comment by CJ-Wright** [comment]
Key questions:
1. Should we stuff the descriptions somewhere?
2. Should we report event sources in the description?

**Comment by CJ-Wright** [comment]
@NSLS-II/dama would it be possible to get comments on this?

**Comment by CJ-Wright** [comment]
It seems that most of the errors are server errors.

**Comment by CJ-Wright** [comment]
@tacaswell where do you want the comment?

**Comment by danielballan** [comment]
Rebased and updated to move release notes to `v0.13.2`, the upcoming release.

**Comment by danielballan** [comment]
Hmm, _all_ of the client tests failed.

**Comment by danielballan** [comment]
To be precise, 3.7 had one random failure (server startup)

```
=== 561 passed, 8 skipped, 2 xfailed, 95 warnings, 1 error in 181.75 seconds ===
```

but on 3.6 the server _always_ failed to start. Re-running....

File: doc/source/whats_new/v0.13.2.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.. currentmodule:: databroker
+
+v0.12.3 (2019-04-22)
```
Comment (mrakitin) [inline doc/source/whats_new/v0.13.2.txt]: Should it match the file name?
```suggestion
v0.13.2 (2019-07-30)
```

File: doc/source/whats_new/v0.13.2.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+--------------------
+
+Bug Fixes
++++++++++
+
+* Support round trip of databroker configs, reporting the config, module and class
```
Comment (mrakitin) [inline doc/source/whats_new/v0.13.2.txt]: Should we also add https://github.com/bluesky/databroker/pull/424 if it goes in and matters?

Comment (danielballan) [reply doc/source/whats_new/v0.13.2.txt]: Yes, but I figured in a separate PR (maybe in #424 itself?)

File: doc/source/whats_new/v0.13.2.txt (hunk 1 above)
Comment (danielballan) [reply doc/source/whats_new/v0.13.2.txt]: Good catch! I don't have permission to accept the change so I'll fix it manually.

**Comment by danielballan** [comment]
The 3.6 build fully passed on the second try.

File: doc/source/whats_new/v0.13.2.txt (hunk 2 above)
Comment (mrakitin) [reply doc/source/whats_new/v0.13.2.txt]: Agreed via DM in Slack to skip mentioning the update from #424 as it does not affect the end users.

**Comment by danielballan** [comment]
Thanks again for this, @CJ-Wright.

**Pull request merged**